aws_advanced_python_wrapper/host_selector.py

```diff
@@ -188,3 +188,61 @@ def _update_cache_properties_for_round_robin_cluster_info(self, round_robin_clus
 
     def clear_cache(self):
         RoundRobinHostSelector._round_robin_cache.clear()
+
+
+class WeightedRandomHostSelector(HostSelector):
+    _DEFAULT_WEIGHT: int = 1
+    _HOST_WEIGHT_PAIRS_PATTERN = r"((?P<host>[^:/?#]*):(?P<weight>.*))"
+    _host_weight_map: Dict[str, int] = {}
+
+    def get_host(self, hosts: Tuple[HostInfo, ...], role: HostRole, props: Optional[Properties] = None) -> HostInfo:
+
+        eligible_hosts: List[HostInfo] = [host for host in hosts if host.role == role and host.get_availability() == HostAvailability.AVAILABLE]
+        eligible_hosts.sort(key=lambda host: host.host, reverse=False)
+        if len(eligible_hosts) == 0:
+            raise AwsWrapperError(Messages.get_formatted("HostSelector.NoHostsMatchingRole", role))
+
+        self._update_host_weight_map_from_string()
+
+        default_weight: int = RoundRobinHostSelector._DEFAULT_WEIGHT
```

Comment:
```suggestion
        default_weight: int = WeightedRandomHostSelector._DEFAULT_WEIGHT
```
same for all other occurrences in this class